src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/BitbucketCloudApiFactory.java

```diff
@@ -12,7 +13,7 @@
 public class BitbucketCloudApiFactory extends BitbucketApiFactory {
     @Override
     protected boolean isMatch(@Nullable String serverUrl) {
-        return serverUrl == null;
+        return serverUrl == null || BitbucketCloudEndpoint.SERVER_URL.equals(serverUrl);
```

Comment:
Using `BitbucketCloudEndpoint.SERVER_URL.equals(serverUrl)` makes it hard to test using wire mock. Can we let consumer of this API pass the cloud endpoint url, be it `https://api.bitbucket.org` or https://bitbucket.org` or a wire mock server `http://localhost:5776`?